src/main/java/com/cloudant/client/api/views/Key.java

```diff
@@ -0,0 +1,152 @@
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.client.api.views;
+
+import com.cloudant.client.internal.views.ComplexKeyImpl;
+
+/**
+ * Class for specifying the types of keys emitted by a view.
+ * <P>Provides static utilities for generating
+ * {@link com.cloudant.client.api.views.Key.ComplexKey} instances.
+ * </P>
+ * <P>Provides access to
+ * {@link com.cloudant.client.api.views.Key.Type}.
+ * </P>
+ */
+public class Key {
+
+    /**
+     * Key type identifier.
+     * <P>
+     * Accessed via the constants:
+     * </P>
+     * <UL>
+     * <LI>{@link com.cloudant.client.api.views.Key.Type#BOOLEAN}</LI>
+     * <LI>{@link com.cloudant.client.api.views.Key.Type#COMPLEX}</LI>
+     * <LI>{@link com.cloudant.client.api.views.Key.Type#NUMBER}</LI>
+     * <LI>{@link com.cloudant.client.api.views.Key.Type#STRING}</LI>
+     * </UL>
+     * New complex keys are initially created using the static helper methods:
+     * <UL>
+     * <LI>{@link com.cloudant.client.api.views.Key#complex(Boolean...)}</LI>
+     * <LI>{@link com.cloudant.client.api.views.Key#complex(Number...)}</LI>
+     * <LI>{@link com.cloudant.client.api.views.Key#complex(String...)}</LI>
+     * </UL>
+     * After creation further values can be added to the complex keys. For example to generate
+     * the complex key {@code [true,1,"a","b"]}
+     * <P>
+     * {@code
+     * Key.complex(true).add(1).add("a", "b");
+     * }
+     * </P>
+     *
+     * @param <T> {@link Boolean},
+     *            {@link Number},
+     *            {@link String} or
+     *            {@link com.cloudant.client.api.views.Key.ComplexKey}
+     */
+    public static final class Type<T> {
+
+        /**
+         * Type constant for boolean keys.
+         */
+        public static final Type<Boolean> BOOLEAN = new Type<Boolean>(Boolean.class);
+        /**
+         * Type constant for
+         * <a href="https://docs.cloudant.com/creating_views.html#map-function-examples">
+         * complex keys.</a>
+         */
+        public static final Type<ComplexKey> COMPLEX = new Type<ComplexKey>(ComplexKey.class);
+        /**
+         * Type constant for numerical keys.
+         */
+        public static final Type<Number> NUMBER = new Type<Number>(Number.class);
+        /**
+         * Type constant for string keys.
+         */
+        public static final Type<String> STRING = new Type<String>(String.class);
+
+        private final Class<T> type;
+
+        private Type(Class<T> type) {
+            this.type = type;
+        }
+
+        Class<T> getType() {
+            return type;
+        }
+    }
+
+    /**
+     * Provides methods for adding values to build a complex key array.
+     */
+    public interface ComplexKey {
```

Comment:
Do we really need this interface? From what I can tell from reading the code and documentation, a complex key, is an array of json values, and The `ComplexKeyImpl` class handles that. From the current implemntation, I don't see what value a user could add, espcially since, the `toString` method is being overriden to provide the JSON string.